webapp/src/DOMJudgeBundle/Service/ScoreboardService.php

```diff
@@ -357,6 +358,35 @@ public function calculateScoreRow(Contest $contest, Team $team, Problem $problem
             }
         }
 
+        // See if this submission was the first to solve this problem
+        // (only relevant if it was correct in the first place)
+        $solvedFirst = false;
+        if ($correctJury) {
+            $params = [
+                ':cid' => $contest->getCid(),
+                ':probid' => $problem->getProbid(),
+                ':teamSortOrder' => $team->getCategory()->getSortorder(),
+                ':submitTime' => $absSubmitTime,
+                ':correctResult' => Judging::RESULT_CORRECT,
+            ];
+
+            // Find out how many valid submissions were submitted earlier
+            // that have a valid judging that is correct, or are awaiting judgement.
+            // Only if there are 0 found, we are definitely the first to solve this problem.
+            $solvedFirst = 0 == $this->entityManager->getConnection()->fetchColumn('
+                SELECT count(*) FROM submission s
+                    LEFT JOIN judging j USING (submitid)
+                    LEFT JOIN team t USING(teamid)
+                    LEFT JOIN team_category tc USING (categoryid)
+                WHERE s.valid = 1 AND
+                    ((j.valid = 1 AND ( j.rejudgingid IS NULL AND (j.result IS NULL OR j.result = :correctResult))) OR
+                      s.judgehost IS NULL) AND
+                    s.cid = :cid AND s.probid = :probid AND
+                    tc.sortorder = :teamSortOrder AND
+                    round(s.submittime,4) < :submitTime',
```

Comment:
why not use the submitid instead?

Reply:
When we import submissions we don’t have any guarantees that submission ID’s and times have any correlation, right? Or am I forgetting something?

Reply:
I reasoned based on the semantics. But indeed if we can keep the guarantee that submitid is ordered it would make it simpler.